pkg/ali/components/DiskGroup.vue

```diff
@@ -1,159 +1,125 @@
-<script>
+<script setup lang="ts">
 import { ref, watch, computed } from 'vue';
 import debounce from 'lodash/debounce';
 import { _EDIT, _VIEW, _CREATE } from '@shell/config/query-params';
 import { removeAt } from '@shell/utils/array';
 import { clone } from '@shell/utils/object';
 import DiskType from './DiskType.vue';
+import { DEFAULT_DISK_VALUE } from '../util/shared';
 
-const defaultAddValue = {
-  category:  'cloud_essd',
-  size:      40,
-  encrypted: 'false'
-};
-
-export default {
-  emits: ['add', 'remove', 'update:value'],
-
-  components: { DiskType },
-  props:      {
-    value: {
-      type:    Array,
-      default: null,
-    },
-    mode: {
-      type:    String,
-      default: _EDIT,
-    },
-    removeAllowed: {
-      type:    Boolean,
-      default: true,
-    },
-    addDisabled: {
-      type:    Boolean,
-      default: false,
-    },
-    loading: {
-      type:    Boolean,
-      default: false
-    },
-    disabled: {
-      type:    Boolean,
-      default: false
-    },
-    options: {
-      type:    Array,
-      default: () => []
-    },
+const emit = defineEmits(['add', 'remove', 'update:value']);
+const props = defineProps({
+  value: {
+    type:    Array,
+    default: null,
+  },
+  mode: {
+    type:    String,
+    default: _EDIT,
+  },
+  removeAllowed: {
+    type:    Boolean,
+    default: true,
+  },
+  addDisabled: {
+    type:    Boolean,
+    default: false,
+  },
+  loading: {
+    type:    Boolean,
+    default: false
   },
-  setup(props, { emit }) {
-    const input = (Array.isArray(props.value) ? props.value : []).slice();
-    const rows = ref([]);
+  disabled: {
+    type:    Boolean,
+    default: false
+  },
+  options: {
+    type:    Array,
+    default: () => []
+  },
+});
```

Comment:
nit: prefer the type-based declaration for better typescript support. We can use `widthDefaults` or reactive props destructuring to assign default values[^1].

[^1]: https://vuejs.org/guide/typescript/composition-api#props-default-values